src/wh_client_cryptocb.c

```diff
@@ -849,4 +875,157 @@ int wh_Client_CryptoCb(int devId, wc_CryptoInfo* info, void* inCtx)
+                         whPacket* packet)
+{
+    int              ret               = 0;
+    whClientContext* ctx               = inCtx;
+    wc_Sha256*       sha256            = info->hash.sha256;
+    uint8_t*         sha256BufferBytes = (uint8_t*)sha256->buffer;
+
+    /* Caller invoked SHA Update:
+     * wc_CryptoCb_Sha256Hash(sha256, data, len, NULL) */
+    if (info->hash.in != NULL) {
+        size_t i = 0;
+
+        /* Process the partial blocks directly from the input data. If there is
+         * enough input data to fill a full block, transfer it to the server */
+        if (sha256->buffLen > 0) {
+            while (i < info->hash.inSz &&
+                   sha256->buffLen < WC_SHA256_BLOCK_SIZE) {
+                sha256BufferBytes[sha256->buffLen++] = info->hash.in[i++];
+            }
+            if (sha256->buffLen == WC_SHA256_BLOCK_SIZE) {
+                ret = _xferSha256BlockAndUpdateDigest(ctx, sha256, packet, 0);
+                sha256->buffLen = 0;
+            }
+        }
+
+        /* Process as many full blocks from the input data as we can */
+        while ((info->hash.inSz - i) >= WC_SHA256_BLOCK_SIZE) {
+            XMEMCPY(sha256BufferBytes, info->hash.in + i, WC_SHA256_BLOCK_SIZE);
+            ret = _xferSha256BlockAndUpdateDigest(ctx, sha256, packet, 0);
+            i += WC_SHA256_BLOCK_SIZE;
+        }
+
+        /* Copy any remaining data into the buffer to be sent in a subsequent
+         * call when we have enough input data to send a full block */
+        while (i < info->hash.inSz) {
+            sha256BufferBytes[sha256->buffLen++] = info->hash.in[i++];
+        }
+    }
+
+    /* Caller invoked SHA finalize:
+     * wc_CryptoCb_Sha256Hash(sha256, NULL, 0, * hash) */
+    if (info->hash.digest != NULL) {
+        ret = _xferSha256BlockAndUpdateDigest(ctx, sha256, packet, 1);
+
+        /* Copy out the final hash value */
+        if (ret == 0) {
+            memcpy(info->hash.digest, sha256->digest, WC_SHA256_DIGEST_SIZE);
+        }
+
+        /* reset the state of the sha context (without blowing away devId) */
+        sha256->buffLen = 0;
+        sha256->flags   = 0;
+        sha256->hiLen   = 0;
+        sha256->loLen   = 0;
+        memset(sha256->digest, 0, sizeof(sha256->digest));
+    }
+
+    return ret;
+}
+
+static int _xferSha256BlockAndUpdateDigest(whClientContext* ctx,
+                                           wc_Sha256* sha256, whPacket* packet,
+                                           uint32_t isLastBlock)
+{
+    uint16_t                   group  = WH_MESSAGE_GROUP_CRYPTO;
+    uint16_t                   action = WH_MESSAGE_ACTION_NONE;
+    int                        ret    = 0;
+    uint16_t                   dataSz = 0;
+    wh_Packet_hash_sha256_req* req    = &packet->hashSha256Req;
+
+    /* Unused wolfCrypt hash flag value. TODO: use devCtx? */
+    const uint32_t RESUME_HASH_BIT_FLAG = 0x80000000;
```

Comment:
Double check that there is case where hiLen/loLen > BLOCK_SIZE but RESUME is not set yet. Seems like we can simplify the compile and logic here if not.